### ISSUE TITLE
Ampere bfloat16 support

### ISSUE THREAD

**Issue by LucasSloan** [tensorflow/tensorflow#47362]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.1/built from head
- Are you willing to contribute it (Yes/No): If you point me at the necessary code, I'll take a crack at it.

**Describe the feature and the current behavior/state.**
I attempted to use bfloat16 training by setting:

```
tf.keras.mixed_precision.set_global_policy('mixed_bfloat16')
```

I got this error message:

```
tensorflow.python.framework.errors_impl.InvalidArgumentError: No OpKernel was registered to support Op 'Conv2D' used by {{node model/conv2d/Conv2D}} with these attrs: [dilations=[1, 1, 1, 1], T=DT_BFLOAT16, data_format="NHWC", strides=[1, 2, 2, 1], use_cudnn_on_gpu=true, explicit_paddings=[], padding="SAME"]
Registered devices: [CPU, GPU]
Registered kernels:
  device='XLA_CPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_BFLOAT16, DT_HALF]
  device='XLA_GPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_BFLOAT16, DT_HALF]
  device='GPU'; T in [DT_INT32]
  device='GPU'; T in [DT_DOUBLE]
  device='GPU'; T in [DT_FLOAT]
  device='GPU'; T in [DT_HALF]
  device='CPU'; T in [DT_INT32]
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_HALF]
```

I read the [cuDNN release notes](https://docs.nvidia.com/deeplearning/cudnn/release-notes/rel_8.html#rel-810) which said that support for bfloat16 came with cuDNN version 8.1, so I built tensorflow from scratch, using CUDA 11.2 and cuDNN 8.1.0.77.

However, I got the same error message.  I've tried this on a trival example (the fashion mnist tutorial) with the same result, so it's not some advanced kernel that I'm using in my code.

I am a little confused that I'm getting this error message even when running with XLA, since the error message says that XLA_GPU_JIT supports DT_BFLOAT16.

**Will this change the current api? How?**
No, you can already specify a bfloat16 mixed precision policy.
**Who will benefit with this feature?**
Anyone with Ampere gpus who wants the speed of 16 bit floats and the numerical stability of bfloat16.

**Comment by czmrand** [comment]
Is there an ETA for this feature request?

**Comment by lightb0x** [comment]
For reference, result of [mixed precision tutorial](https://www.tensorflow.org/guide/mixed_precision) on RTX 3070 is as follows.
* TF32 -> 0.9116, ~70ms per step (default)
* FP16 -> 0.9274, ~45ms per step
* BF16 -> 0.9477, ~3000ms per step

If model doesn't contain convolution layer, it runs correctly, even if it's slow. 

(tested environment : Ubuntu 20.04, [nvidia tensorflow container image 21.03](https://docs.nvidia.com/deeplearning/frameworks/tensorflow-release-notes/rel_21-04.html#rel_21-04))

**Comment by tilakrayal** [comment]
@LucasSloan,
There was a related PR(#59972) which was raised for the **bfloat16** has been merged and also I tried to execute the sample code on tensorflow v2.15 (GPU) by importing the keras from the Keras 3 & the code was executed without any issue/error. Kindly find the gist of it here(https://colab.research.google.com/gist/tilakrayal/fdc27fa1701f846a63b859b9dab4132c/untitled1575.ipynb). Thank you!
